source/common/tracing/http_tracer_impl.cc

```diff
@@ -224,6 +221,23 @@ void HttpTracerUtility::setCommonTags(Span& span, const Http::HeaderMap* respons
   }
 }
 
+CustomTagConstSharedPtr
+HttpTracerUtility::createCustomTag(const envoy::type::tracing::v2::CustomTag& tag) {
+  switch (tag.type_case()) {
+  case envoy::type::tracing::v2::CustomTag::kLiteral:
+    return std::make_shared<Tracing::LiteralCustomTag>(tag.tag(), tag.literal());
+  case envoy::type::tracing::v2::CustomTag::kEnvironment:
+    return std::make_shared<Tracing::EnvironmentCustomTag>(tag.tag(), tag.environment());
+  case envoy::type::tracing::v2::CustomTag::kRequestHeader:
+    return std::make_shared<Tracing::RequestHeaderCustomTag>(tag.tag(), tag.request_header());
+  case envoy::type::tracing::v2::CustomTag::kMetadata:
+    return std::make_shared<Tracing::MetadataCustomTag>(tag.tag(), tag.metadata());
+  case envoy::type::tracing::v2::CustomTag::TYPE_NOT_SET:
```

Comment:
This is checked by schema. Just add a default case and use NOT_REACHED, same below if needed for the compiler.